tools/customization_dataset_preparation/customization_dataset_preparation.py

```diff
@@ -0,0 +1,331 @@
+    return None
+
+
+def show_first_example_in_df(df):
+    message = ''
+    for column in df.columns:
+        # prints \n instead of an a newline
+        column_value = df[column][0].replace('\n', '\\n')
+        message += f"-->Column {column}:\n{column_value}\n"
+    return message
+
+
+def get_prepared_filename(filename, split_train_validation=False):
+    message = ""
+    if not split_train_validation:
+        new_filename = filename.replace(".jsonl", "_prepared.jsonl")
+        retry = 0
+        while os.path.isfile(new_filename):
+            message += f"File {new_filename} exists. Trying next available filename increment\n"
+            retry += 1
+            new_filename = filename.replace(".jsonl", f"_prepared{retry}.jsonl")
+        return new_filename, message
+    else:
+        train_filename = filename.replace(".jsonl", "_prepared_train.jsonl")
+        val_filename = filename.replace(".jsonl", "_prepared_val.jsonl")
+        retry = 0
+        while os.path.isfile(train_filename) or os.path.isfile(val_filename):
+            message += f"File {train_filename} or {val_filename} exists. Trying next available filename increment\n"
+            retry += 1
+            train_filename = filename.replace(".jsonl", f"_prepared_train{retry}.jsonl")
+            val_filename = filename.replace(".jsonl", f"_prepared_val{retry}.jsonl")
+        return [train_filename, val_filename], message if message else None
+
+
+def split_into_train_validation(df, val_proportion=0.1):
+    n_val = int(val_proportion * len(df))
+    df_val = df.sample(n=n_val, random_state=42)
+    df_train = df.drop(df_val.index)
+    return df_train, df_val
+
+
+def write_df_to_jsonl(df, filename):
+    df.to_json(filename, lines=True, orient="records", force_ascii=False)
+    return f"File {filename} written"
+
+
+def print_select_messages(title, select_messages):
+    print("*" * 40)
+    print(title)
+    print("*" * 40)
+    for idx, message in enumerate(select_messages):
+        print(f"{idx+1}.")
+        print(message)
+
+
+def print_all_messages(messages):
+    messages = [message for message in messages if message]
+    info_messages = [message for message in messages if not message.startswith("TODO")]
+    to_do_messages = [message for message in messages if message.startswith("TODO")]
+
+    print_select_messages("ACTIONABLE MESSAGES", to_do_messages)
+    print_select_messages("INFORMATIONAL MESSAGES", info_messages)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Prepares data for NeMoLLM Csutomization Service")
+    parser.add_argument("--filename", "-f")
+    parser.add_argument("--prompt_template", "-pt", default="{prompt}")
+    parser.add_argument("--completion_template", "-ct", default="{completion}")
+    parser.add_argument("--long_seq_model", "-8k", action="store_true")
```

Comment:
should be 2k by default

Reply:
Yes default behavior will be 2k (8k only when user specifies --long_seq_model or -8k)